map.go

```diff
@@ -460,6 +439,31 @@ func (spec *MapSpec) createMap(inner *sys.FD, opts MapOptions) (_ *Map, err erro
 		if attr.BtfFd == 0 {
 			return nil, fmt.Errorf("map create: %w (without BTF k/v)", err)
 		}
+
+		if spec.Flags&(unix.BPF_F_RDONLY_PROG|unix.BPF_F_WRONLY_PROG) > 0 || spec.Freeze {
```

Comment:
I think this needs to move before `if attr.BtfFd == 0`: otherwise we never execute these conditions if a map is created without BTF.

Reply:
Good catch

Reply:
But then again, even if it comes above it ,I reckon eventually it will still enter the `if attr.BtfFd == 0 {}` block. Is there a reason why we'd prefer to throw the error at `if spec.Flags&(unix.BPF_F_RDONLY_PROG|unix.BPF_F_WRONLY_PROG) > 0 || spec.Freeze` instead?

Reply:
Not sure if that's what you meant but decided to move the `btf` check as the fallback error check because it could mask potentially more prevalent errors which arise from the feature tests.